torch/utils/cpp_extension.py

```diff
@@ -259,8 +308,8 @@ def build_extensions(self):
             self._define_torch_extension_name(extension)
             self._add_gnu_cpp_abi_flag(extension)
 
-        # Register .cu and .cuh as valid source extensions.
-        self.compiler.src_extensions += ['.cu', '.cuh']
+        # Register .cu, .cuh and .hip as valid source extensions.
```

Comment:
(putting this here because github won't let me put it elsewhere)

We should toss a check for use_ninja to `BuildExtension.__init__`. If `use_ninja` is True and we are building a ROCm extension, we should error out gracefully so that ROCm users aren't confused about the extension not building. (I am not sure when we'll be able to build ROCm extensions with ninja; I know that's next on your list but this check is for just in case we don't get to that by the next release).

Reply:
Thank you for the catch. Just added a fallback in the `__init__` similar to the behavior when ninja is not present in the system